src/main/java/seedu/duke/commands/UpdateCommand.java

```diff
@@ -26,60 +35,76 @@ public UpdateCommand(String arguments) throws DukeException {
         this.flags = Arrays.copyOfRange(details, 1, details.length);
     }
 
+    /**
+     * A core logic method for updating the current food attributes with given flag values
+     *
+     * @param currentFood
+     * @param flagName
+     * @param flagValue
+     * @throws DukeException
+     */
+    private void updateFoodAttribute(Food currentFood, String flagName, String flagValue) throws DukeException {
+        try {
+            switch (flagName) {
+            case "n":
+                currentFood.setName(flagValue);
+                break;
+            case "e":
+                Validator.isExpiryDateValid(flagValue);
+                currentFood.setExpiryDate(flagValue);
+                break;
+            case "q":
+                if (currentFood.getQuantity() == 0 && Arrays.asList(flags).contains("u")) {
+                    throw new DukeException("Can't set quantity with no unit provided");
+                }
+                currentFood.setQuantity(Double.parseDouble(flagValue));
+                String currentFoodUnit = currentFood.getUnit();
+                currentFood.setUnit(currentFoodUnit);
+                break;
+            case "u":
+                currentFood.setUnit(flagValue);
```

Comment:
i find ^^ this one quite important to change as it might be regarded as a bug to user.